dotCMS/src/main/java/com/dotmarketing/common/db/DotDatabaseMetaData.java

```diff
@@ -0,0 +1,315 @@
+                            // constraint. Add the column to the same object
+                            // instead of duplicating an entry in the list.
+                            ForeignKey existingKey = foreignKeyMap.get(foreignKey.fkName());
+                            existingKey.addPrimaryColumnName(pkColumnName);
+                            existingKey.addForeignColumnName(fkColumnName);
+                        }
+                    }
+
+                }
+            }
+        } catch (SQLException e) {
+            Logger.error(this,
+                    "An error occurred when getting the the foreign keys: " + e.getMessage(), e);
+        }
+
+        return Collections.unmodifiableList(foreignKeys);
+    } // getForeignKeys.
+
+    /**
+     * Drops a foreign key
+     * @param foreignKey {@link ForeignKey}
+     * @throws SQLException
+     */
+    public void dropForeignKey (final ForeignKey foreignKey) throws SQLException {
+
+        this.dropForeignKey(DbConnectionFactory.getConnection(), foreignKey);
+    } // dropForeignKey.
+
+    /**
+     * Drops a foreign key
+     * @param conn {@link Connection}
+     * @param foreignKey {@link ForeignKey}
+     * @throws SQLException
+     */
+    public void dropForeignKey (final Connection conn, final ForeignKey foreignKey) throws SQLException {
+        if(DbConnectionFactory.isPostgres() ||
+                DbConnectionFactory.isMsSql() ||
+                DbConnectionFactory.isOracle()) {
+
+            this.executeDropConstraint(conn, foreignKey.getForeignKeyTableName(), foreignKey.getForeignKeyName());
+        } else if (DbConnectionFactory.isMySql()) {
+            executeDropForeignKeyMySql(conn, foreignKey.getForeignKeyTableName(), foreignKey.getForeignKeyName());
+        }
+    } // dropForeignKey.
+
+    /**
+     * Drops the specified constraint associated to the given table.
+     *
+     * @param conn
+     *            - The database connection object to access the dotCMS data.
+     * @param tableName
+     *            - The name of the table whose constraint will be dropped.
+     * @param constraintName
+     *            - The name of the constraint that will be dropped.
+     * @throws SQLException
+     *             An error occurred when executing the SQL query.
+     */
+    public void executeDropConstraint(final Connection conn, final String tableName, final String constraintName) throws SQLException {
+
+        String sql = StringPool.BLANK;
+        PreparedStatement preparedStatement = null;
+
+        if(DbConnectionFactory.isMySql()) {
+            sql = (constraintName.indexOf("PRIMARY")>-1)?
+                ALTER_TABLE + tableName + " DROP PRIMARY KEY ":
+                ALTER_TABLE + tableName + " DROP INDEX " + constraintName;
+        }  else {
+            sql = ALTER_TABLE + tableName + " DROP CONSTRAINT " + constraintName;
+        }
+
+        try {
+            preparedStatement = conn.prepareStatement(sql);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use a variable binding mechanism to construct this query instead of concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2077)